### PR TITLE
Update PHP Snippet generation.

### PR DESCRIPTION
**Pull request by SilasKenneth**
## Overview

Brief description of what this PR does, and why it is needed.

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* How to test this PR
* Prefer bulleted description
* Start after checking out this branch
* Include any setup required, such as bundling scripts, restarting services, etc.
* Include test case, and expected output

### CODE REVIEW

**Review by Ndiritu — COMMENTED** [review]
Would it confuse customers if query parameters snippets only work with PHP 8 while the others work across both PHP 7 and 8?

**Comment by SilasKenneth** [comment]
> Would it confuse customers if query parameters snippets only work with PHP 8 while the others work across both PHP 7 and 8?

That makes sense. What about having something like this? Knowing these other query parameters that are supposed to be set to null is the problem, at least on the devx side

```php
...addQueryParameters(
null,
null,
'age eq 20',
['email', 'principalName']);

```
Won't that be confusing, too?

**Comment by Ndiritu** [comment]
> 

I think for PHP 7 devs the existing snippets seem like the best experience. PHP 8, constructors would be ideal.
Unless we add both snippets? Or only add a comment in the snippet about a better experience for PHP 8 using the constructors?

File: CodeSnippetsReflection.OpenAPI.Test/PhpGeneratorTests.cs [hunk 1]
```diff
@@ -83,8 +83,8 @@ public async Task GeneratesObjectInitializationWithCallToSetters()
             };
         var snippetModel = new SnippetModel(requestPayload, ServiceRootUrl, await GetV1TreeNode());
         var result = _generator.GenerateCodeSnippet(snippetModel);
-        Assert.Contains("$queryParameters = new UsersRequestBuilderPostQueryParameters();", result);
-        Assert.Contains("$requestConfiguration->queryParameters = $queryParameters;", result);
+        Assert.Contains("$requestConfiguration = new UsersRequestBuilderPostRequestConfiguration();", result);
+        Assert.Contains("$queryParameters = UsersRequestBuilderPostRequestConfiguration::createQueryParameters();", result);
```
Comment (Ndiritu) [inline CodeSnippetsReflection.OpenAPI.Test/PhpGeneratorTests.cs:87]: 👍🏾

**Pull request merged**